cms/djangoapps/contentstore/views/program.py

```diff
@@ -38,3 +39,16 @@ def get(self, request, *args, **kwargs):
             })
         else:
             raise Http404
+
+
+class ProgramsIdTokenView(View):
+    """Provides id tokens to JavaScript clients for use with the Programs API."""
+
+    @method_decorator(login_required)
```

Comment:
Does this work? Django's docs on class-based views recommend decorating the [dispatch method](https://docs.djangoproject.com/en/1.9/topics/class-based-views/intro/#decorating-the-class).